### PR TITLE
haskell: Use stack instead of cabal

### PR DESCRIPTION
**Pull request by gfontenot**
:fire:

### CODE REVIEW

File: tag-haskell/setup/setup [hunk 1]
```diff
@@ -5,9 +5,5 @@ set -e
 echo "\n==========================================="
 echo "Setting up Haskell\n"
 
-echo "Installing and updating cabal"
-cabal install cabal-install
-cabal update
-
 echo "Installing basic dependencies"
-cabal install alex happy yesod-bin hoogle
+stack install alex happy yesod-bin hoogle
```
Comment (gfontenot) [inline tag-haskell/setup/setup]: not sure if this makes sense anymore

File: tag-haskell/config/haskell/functions.zsh [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-function ghc-pkg-clean() {
-  for p in `ghc-pkg check $* 2>&1  | grep problems | awk '{print $6}' | sed -e 's/:$//'`
-  do
-    echo unregistering $p; ghc-pkg $* unregister $p
-  done
-}
-
-# remove all installed GHC/cabal packages, leaving ~/.cabal binaries and docs in place.
-# When all else fails, use this to get out of dependency hell and start over.
-function fuck-off-cabal() {
```
Comment (gabebw) [inline tag-haskell/config/haskell/functions.zsh]: ha

File: tag-haskell/ghci [hunk 3]
```diff
@@ -11,5 +11,5 @@ import Data.Char
 import Control.Applicative
 import Control.Monad
 
--- cabal install hoogle, then `:hoogle fmap`
+-- stack install hoogle, then `:hoogle fmap`
 :def hoogle \s -> return $ ":! hoogle --count=15 \"" ++ s ++ "\""
```
Comment (gabebw) [inline tag-haskell/ghci]: Is this from me?!

Comment (gfontenot) [inline tag-haskell/ghci]: no i've never stolen anything from your dotfiles and I don't appreciate the implication

**Pull request merged**